lib/ocpp/v201/charge_point.cpp

```diff
@@ -470,27 +470,76 @@ void ChargePoint::init_websocket() {
-            this->message_queue->pause();
-
-            if (!this->disable_automatic_websocket_reconnects) {
-                this->websocket_timer.timeout(
-                    [this, reason]() {
-                        if (reason != websocketpp::close::status::service_restart) {
-                            this->next_network_configuration_priority();
+        if (this->registration_status == RegistrationStatusEnum::Accepted) {
+            // handle offline threshold
+            //  Get the current time point using steady_clock
+            std::chrono::steady_clock::duration offline_duration = std::chrono::steady_clock::now() - time_disconnected;
+
+            // B04.FR.01
+            // If offline period exceeds offline threshold then send the status notification for all connectors
+            if (offline_duration > (std::chrono::seconds)this->device_model->get_value<int>(
+                                       ControllerComponentVariables::OfflineThreshold)) {
+                EVLOG_debug << "offline for more than offline threshold ";
+                for (auto const& [evse_id, evse] : this->evses) {
+                    evse->trigger_status_notification_callbacks();
+                }
+            } else {
+                // B04.FR.02
+                // If offline period doesn't exceed offline threshold then send the status notification for all
+                // connectors that changed state
+                EVLOG_debug << "offline for less than offline threshold ";
+                for (auto const& [evse_id, evse] : this->evses) {
+                    int number_of_connectors = evse->get_number_of_connectors();
+                    for (int connector_id = 1; connector_id <= number_of_connectors; connector_id++) {
+                        if (conn_state_per_evse[{evse_id, connector_id}] != evse->get_state(connector_id)) {
+                            evse->trigger_status_notification_callback(connector_id);
                         }
-                        this->start_websocket();
-                    },
-                    WEBSOCKET_INIT_DELAY);
+                    }
+                }
             }
-        });
+        }
+    });
+
+    this->websocket->register_closed_callback([this, connection_options,
+                                               configuration_slot](const websocketpp::close::status::value reason) {
+        EVLOG_warning << "Failed to connect to NetworkConfigurationPriority: "
+                      << this->network_configuration_priority + 1
+                      << " which is configurationSlot: " << configuration_slot;
+        this->websocket_connection_status = WebsocketConnectionStatusEnum::Disconnected;
+        this->message_queue->pause();
+
+        // check if offline threshold has been defined
+        if ((std::chrono::seconds)this->device_model->get_value<int>(ControllerComponentVariables::OfflineThreshold) !=
```

Comment:
Casting both to std::chrono::seconds not necessary here; Could just be 

`this->device_model->get_value<int>(ControllerComponentVariables::OfflineThreshold) != 0`